frontend/src/components/Hotel/Hotel.tsx

```diff
@@ -0,0 +1,140 @@
+import './hotel.css'
+import LoadingSpinner from '../LoadingSpinner';
+import { useHotel } from '../../hooks/useHotel';
+import { MdShoppingBag } from 'react-icons/md';
+import { error } from 'console';
+
+
+function Hotel() {
+  const { hotelDetails,
+    availableRooms,
+    checkIn,
+    checkOut,
+    fullscreen,
+    reviews,
+    loading,
+    cartItems,
+    validationError,
+    setCheckIn,
+    setCheckOut,
+    setFullscreen,
+    handleFetchAvailableRoom,
+    handleAddToCart } = useHotel()
+
+  const ShoppingCartIcon = MdShoppingBag as unknown as React.FC;
+
+  if (loading) return <LoadingSpinner />
+
+  return (
+    <div className="hotel-page">
+      <header className="hotel-header">
+        <button className="cart-button">
+          <ShoppingCartIcon />
+          <span className='cart-items'>{cartItems.length}</span>
+        </button>
+      </header>
+
+      <div className="hotel-content">
+        <aside className="hotel-left">
+          <section className="hotel-details">
+            <h2>{hotelDetails?.hotelName}</h2>
+            <p>⭐ {hotelDetails?.starRating} stars</p>
+            <p>{hotelDetails?.description}</p>
+            <p><strong>Location:</strong> {hotelDetails?.location}</p>
+          </section>
+
+          <section className="hotel-map">
+            <h3>Location on Map</h3>
+            <div className="map-placeholder">
+              {hotelDetails && (
+                <iframe
+                  title="Hotel Location"
+                  className="map-frame"
+                  width="100%"
+                  height="250"
+                  style={{ border: 0 }}
+                  src={`https://www.google.com/maps?q=${hotelDetails.latitude},${hotelDetails.longitude}&z=15&output=embed`}
+                  allowFullScreen
+                ></iframe>
+              )}
+            </div>
+          </section>
+        </aside>
+
+        <main className="hotel-right">
+          <section className="hotel-gallery">
+            <h3>Picture Gallery</h3>
+            <div className={`gallery-placeholder ${fullscreen ? 'fullscreen' : ''}`} onClick={() => setFullscreen(!fullscreen)}>
+              <img src="https://cf.bstatic.com/xdata/images/hotel/max1280x900/33143786.jpg?k=4d0bca9d9795b80beb2cd9786946e043b23d1372eb633d5855d3aba6343d68d4&o=&hp=1" alt={hotelDetails?.hotelName} />
+              <span className="fullscreen-hint">{fullscreen ? 'Exit fullscreen' : 'Click to view fullscreen'}</span>
+            </div>
+          </section>
+
+          <section className="hotel-rooms">
+            <h3>List of Available Rooms</h3>
+            <p>Available Rooms: {hotelDetails?.availableRooms}</p>
+            <div className='date-picker'>
+              <label>
+                <span>Check-in</span>
+                <input type="date" value={checkIn} onChange={(e) => setCheckIn(e.target.value)} name="checkIn" id="checkIn" />
```

Comment:
also, we can do this
```suggestion
                <input type="date" value={checkIn} onChange={handleCheckInChange} name="checkIn" id="checkIn" />
```

and update the handleCheckInChange callback to accept event type